internal/cache/key.go

```diff
@@ -0,0 +1,9 @@
+package cache
+
+import "crypto/sha1"
+
+type chacheKey [sha1.Size]byte
+
+func newCacheKey(v []byte) chacheKey {
+	return sha1.Sum(v)
```

Comment:
Is it important that we use a cryptographic hash? It seems to me the only property we care about here is that the key is unique so we might be better off with something less costly such as the [maphash](https://pkg.go.dev/hash/maphash) or [fnv](https://pkg.go.dev/hash/fnv) from the standard library.

Reply:
I will change it to maphash